multicluster/controllers/multicluster/memberclusterannounce_controller.go

```diff
@@ -114,28 +110,26 @@ func (r *MemberClusterAnnounceReconciler) Reconcile(ctx context.Context, req ctr
 		klog.V(2).InfoS("Reset lastUpdateTime", "cluster", memberAnnounce.ClusterID)
 		// Reset lastUpdateTime and connectedLeader for this member.
 		data.lastUpdateTime = time.Now()
-		if len(memberAnnounce.LeaderClusterID) == 0 {
-			data.leaderStatus.connectedLeader = v1.ConditionUnknown
-			data.leaderStatus.message = "Not connected to leader yet"
-			data.leaderStatus.reason = ""
+		data.leaderStatus.connectedLeader = v1.ConditionFalse
+		data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
+			memberAnnounce.ClusterID)
+		data.leaderStatus.reason = ReasonNotLeader
+		// Check whether this local cluster is the leader for this member.
+		clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
+		if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
+			if clusterClaim.Value == memberAnnounce.LeaderClusterID {
+				data.leaderStatus.connectedLeader = v1.ConditionTrue
+				data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
+					memberAnnounce.ClusterID)
+				data.leaderStatus.reason = ReasonConnectedLeader
+			}
 		} else {
-			data.leaderStatus.connectedLeader = v1.ConditionFalse
-			data.leaderStatus.message = fmt.Sprintf("Local cluster is not the leader of member: %v",
-				memberAnnounce.ClusterID)
-			data.leaderStatus.reason = ReasonNotLeader
-			// Check whether this local cluster is the leader for this member.
-			clusterClaim := &multiclusterv1alpha2.ClusterClaim{}
-			if err := r.Get(context.TODO(), types.NamespacedName{Name: multiclusterv1alpha2.WellKnownClusterClaimID, Namespace: req.Namespace}, clusterClaim); err == nil {
-				if clusterClaim.Value == memberAnnounce.LeaderClusterID {
-					data.leaderStatus.connectedLeader = v1.ConditionTrue
-					data.leaderStatus.message = fmt.Sprintf("Local cluster is the leader of member: %v",
-						memberAnnounce.ClusterID)
-					data.leaderStatus.reason = ReasonConnectedLeader
-				}
+			if !apierrors.IsNotFound(err) {
+				return ctrl.Result{}, err
 			}
-			// If err != nil, probably ClusterClaims were deleted during the processing of MemberClusterAnnounce.
-			// Nothing to handle in this case and MemberClusterAnnounce will also be deleted soon.
-			// TODO: Add ClusterClaim webhook to make sure it cannot be deleted while ClusterSet is present.
+			// If it's not found error, ClusterClaims were probably deleted during the processing of MemberClusterAnnounce.
+			// Nothing to handle in this case since it means no ClusterSet is referring to it or the ClusterSet is deleted
```

Comment:
What does "no ClusterSet is referring to it" mean?

Reply:
I mixed the wrong message. It should say 'since the ClusterSet is deleted'.